src/frontend/context/video-context.js

```diff
@@ -0,0 +1,39 @@
+import { createContext, useContext, useReducer, useEffect } from 'react';
+import { getVideos } from '../utils/video-request';
+
+const VideoContext = createContext();
+
+const reducer = (state, action) => {
+	switch (action.type) {
+		case 'GET_VIDEOS':
+			return {
+				...state,
+				videoList: action.payload
+			};
+		default:
+			return state;
+	}
+};
+
+const VideoProvider = ({ children }) => {
+	const [state, dispatch] = useReducer(reducer, { videoList: [] });
```

Comment:
you can use better naming like videosState and videosDispatch

Reply:
Yes, will make the changes